scripts/configurator/parser/cmake_parser.py

```diff
@@ -1,23 +1,30 @@
 from cmake_row import cmake_row, row_type
-from typing import List
+from typing import Deque, List
+from collections import deque
 
 def read_file(filepath):
    with open(filepath, 'r') as f:
       return f.readlines()
 
 def parse_rows(file_lines: List[str]) -> List[cmake_row]:
    cmake_rows: List[cmake_row] =  []
-   comment = ""
+   comment_list: Deque[str] = deque()
    for i, line in enumerate(file_lines):
       current_row = cmake_row(line, i)
       cmake_rows.append(current_row)
       if current_row.row_type == row_type.VARIABLE:
          j = i-1
          while j >= 0 and cmake_rows[j].row_type == row_type.SLASH_COMMENT:
-            comment = cmake_rows[j].get_val() + comment
+            comment_list.appendleft(cmake_rows[j].get_val() + "\n")
             j -= 1
-         current_row.comment = comment
-         comment = ""
+         """
+         In this case we need a new cmake_row to handle all the new
+         information provided by the metadata, so every cmake_row will be
+         constructed twice. The process could also be extracted to other
+         functions, but this respects the seperation of concerns.
+         """
+         cmake_rows[i] = cmake_row(line, j, comment_list)
```

Comment:
Ok, that makes sense

Reply:
I'm not sure if i understand it correctly. My approach would be of just using a deque converting to list when needed (with `list(comment_list)`). is this correct?

Reply:
No, not when needed. Convert to list just once, before passing it to `cmake_row` and keep it as list.

Reply:
Wait, why do we care about keeping the list? it's the only case in which it will be used, in the other case we just use the default parameter, which is set to empty.